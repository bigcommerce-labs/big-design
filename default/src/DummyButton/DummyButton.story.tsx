import { action } from '@storybook/addon-actions';
import { select } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react';
import * as React from 'react';

import { DummyButton } from './DummyButton';

const variant = ['primary' as 'primary', 'secondary' as 'secondary'];

storiesOf('DummyButton', module)
  .add('Primary', () => (
    <DummyButton onClick={action('click')} variant={select('variant', variant, 'primary')}>
      I'm a button
    </DummyButton>
  ))
  .add('Secondary', () => (
    <DummyButton onClick={action('click')} variant={select('variant', variant, 'secondary')}>
      I'm a button
    </DummyButton>
  ));
