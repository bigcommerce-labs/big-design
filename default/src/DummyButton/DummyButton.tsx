import styled, { css } from 'styled-components';

interface Props {
  variant: 'primary' | 'secondary';
}

export const DummyButton = styled('button')<Props>`
  color: white;

  ${props =>
    props.variant === 'primary' &&
    css`
      background-color: blue;
    `}

  ${props =>
    props.variant === 'secondary' &&
    css`
      background-color: red;
      border-radius: 10px;
    `}
`;
