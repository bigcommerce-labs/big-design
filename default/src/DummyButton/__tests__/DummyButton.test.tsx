import 'jest-styled-components';
import * as React from 'react';
import renderer from 'react-test-renderer';

import { DummyButton } from '../DummyButton';

describe('DummyButton', () => {
  test('dummy test', () => {
    const button = renderer.create(<DummyButton variant="primary">Click me</DummyButton>);

    const tree = button.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
