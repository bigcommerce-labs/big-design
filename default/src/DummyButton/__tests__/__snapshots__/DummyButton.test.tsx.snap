// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DummyButton dummy test 1`] = `
.c0 {
  color: white;
  background-color: blue;
}

<button
  className="c0"
>
  Click me
</button>
`;
